core/include/traccc/ambiguity_resolution/legacy/greedy_ambiguity_resolution_algorithm.hpp

```diff
@@ -0,0 +1,140 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// System include
+#include <algorithm>
+#include <initializer_list>
+#include <iostream>
+#include <limits>
+#include <map>
+#include <set>
+#include <unordered_map>
+#include <vector>
+
+// VecMem include(s).
+#include <vecmem/containers/vector.hpp>
+
+// Project include(s).
+#include "traccc/definitions/qualifiers.hpp"
+#include "traccc/edm/track_candidate.hpp"
+#include "traccc/edm/track_state.hpp"
+#include "traccc/utils/algorithm.hpp"
+#include "traccc/utils/messaging.hpp"
+
+// Greedy ambiguity resolution adapted from ACTS code
+
+namespace traccc::legacy {
+
+/// Evicts tracks that seem to be duplicates or fakes. This algorithm takes a
+/// greedy approach in the sense that it will remove the track which looks "most
+/// duplicate/fake" first and continues the same process with the rest. That
+/// process continues until the final state conditions are met.
+///
+/// The implementation works as follows:
+///  1) Calculate shared hits per track.
+///  2) If the maximum shared hits criteria is met, we are done.
+///     This is the configurable amount of shared hits we are ok with
+///     in our experiment.
+///  3) Else, remove the track with the highest relative shared hits (i.e.
+///     shared hits / hits).
+///  4) Back to square 1.
+class greedy_ambiguity_resolution_algorithm
+    : public algorithm<track_candidate_container_types::host(
+          const typename track_candidate_container_types::host&)>,
+      public messaging {
+
+    public:
+    struct config_t {
+
+        config_t(){};
+
+        /// Maximum amount of shared hits per track. One (1) means "no shared
+        /// hit allowed".
+        std::uint32_t maximum_shared_hits = 1;
+
+        /// Maximum number of iterations.
+        std::uint32_t maximum_iterations = 1000000;
+
+        /// Minimum number of measurement to form a track.
+        std::size_t n_measurements_min = 3;
+    };
```

Comment:
Isn't this basically the same as `resolution_config` below?

Reply:
Right but I just want to leave the legacy algorithm as is